src/harness.ts

```diff
@@ -65,9 +65,9 @@ function findDNodeByKey(target: DNode, key: string | object): HNode | WNode | un
 function stubRender(target: DNode): DNode {
 	decorate(
 		target,
-		(dNode: WNode) => {
-			const { widgetConstructor, properties } = dNode;
-			dNode.widgetConstructor = StubWidget;
+		(dNode: DNode) => {
```

Comment:
Maybe we could change the signature on decorate to something like this to fix this type?
```
export function decorate<V extends DNode = DNode>(dNodes: DNode, modifier: (dNode: V) => void, predicate?: (dNode: DNode) => dNode is V): DNode;
export function decorate(dNodes: DNode, modifier: (dNode: DNode) => void, predicate?: (dNode: DNode) => dNode is V): DNode;
```

Not that I expect you to do that, just thinking out loud.

Reply:
I got curious and figured out that the signature for `decorate` could be changed to:

```
export function decorate<T extends DNode>(dNodes: DNode, modifier: (dNode: T) => void, predicate: (dNode: DNode) => dNode is T): DNode;
export function decorate<T extends DNode>(dNodes: DNode[], modifier: (dNode: T) => void, predicate: (dNode: DNode) => dNode is T): DNode[];
export function decorate(dNodes: DNode, modifier: (dNode: DNode) => void): DNode;
export function decorate(dNodes: DNode[], modifier: (dNode: DNode) => void): DNode[];
```